File: from-1601-to-1700/1605-find-valid-matrix-given-row-and-column-sums/solution.go
func restoreMatrix(rowSum []int, colSum []int) [][]int {
    m, n := len(rowSum), len(colSum)
    result := make([][]int, m)
    
    for r := 0; r < m; r ++ {
        result[r] = make([]int, n)
        for c := 0; c < n; c ++ {
            result[r][c] = min(rowSum[r], colSum[c])
            rowSum[r] -= result[r][c]
            colSum[c] -= result[r][c]
        }
    }
    return result
}

File: from-1601-to-1700/1605-find-valid-matrix-given-row-and-column-sums/solution.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        m, n  = len(rowSum), len(colSum)
        ans = [[0] * n for _ in range(m)]
        
        for r in range(m):
            for c in range(n):
                ans[r][c] = min(rowSum[r], colSum[c])
                rowSum[r] -= ans[r][c]
                colSum[c] -= ans[r][c]
                
        return ans